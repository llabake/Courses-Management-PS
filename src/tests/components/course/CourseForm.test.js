import React from 'react';
import { shallow } from 'enzyme';
import CourseForm from "../../../components/course/CourseForm";

describe('<CourseForm/>', () => {
  it('should render correctly', () => {
    const props = {
      course: {},
      onSave: jest.fn(),
      onChange: jest.fn(),
      saving: false,
      errors: [],
    };
    const wrapper = shallow(<CourseForm {...props} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('input').props().value).toBe('Save');
  });
  it('should should saving when saving', () => {
    const props = {
      course: {},
      onSave: jest.fn(),
      onChange: jest.fn(),
      saving: true,
      errors: [],
    };
    const wrapper = shallow(<CourseForm {...props} />);
    expect(wrapper.find('input').props().value).toBe('Saving');
  });
});
