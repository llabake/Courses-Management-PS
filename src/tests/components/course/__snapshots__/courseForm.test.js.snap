// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CourseForm/> should render correctly 1`] = `
<form>
  <h1>
    Manage Course
  </h1>
  <TextInput
    label="Title"
    name="title"
    onChange={[MockFunction]}
  />
  <SelectInput
    defaultOption="Select Author"
    label="Author"
    name="authorId"
    onChange={[MockFunction]}
  />
  <TextInput
    label="Category"
    name="category"
    onChange={[MockFunction]}
  />
  <TextInput
    error={0}
    label="Length"
    name="length"
    onChange={[MockFunction]}
  />
  <input
    className="btn btn-primary"
    disabled={false}
    onClick={[MockFunction]}
    type="submit"
    value="Save"
  />
</form>
`;
